synthesis/synthesis.go

```diff
@@ -260,76 +181,61 @@ func FixCds(sqlitePath string, sequence string, codontable codon.Table, problema
 				if codon.Triplet != toCodon.Triplet {
 					toCodonBias := strings.Count(toCodon.Triplet, "G") + strings.Count(toCodon.Triplet, "C")
 					switch {
-					case codonBias == toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "NA")
 					case codonBias > toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "AT")
+						atBiasMap[codon.Triplet] = append(atBiasMap[codon.Triplet], toCodon.Triplet)
 					case codonBias < toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "GC")
+						gcBiasMap[codon.Triplet] = append(gcBiasMap[codon.Triplet], toCodon.Triplet)
 					}
+					naBiasMap[codon.Triplet] = append(naBiasMap[codon.Triplet], toCodon.Triplet)
 				}
 			}
 		}
+		// aminoAcidTotal can't equal zero or else division later will panic
 		if aminoAcidTotal == 0 {
 			aminoAcidTotal = 1
 		}
-		weightTable[aminoAcid.Letter] = aminoAcidTotal
-	}
-
-	// Insert seq and history
-	pos := 0
-	for codonPosition := 0; codonPosition < len(sequence); codonPosition = codonPosition + codonLength {
-		codon := sequence[codonPosition : codonPosition+codonLength]
-		tx.MustExec(`INSERT INTO seq(pos) VALUES (?)`, pos)
-		tx.MustExec(`INSERT INTO history(pos, codon, step) VALUES (?, ?, 0)`, pos, codon)
-		pos++
+		aminoAcidWeightTable[aminoAcid.Letter] = aminoAcidTotal
 	}
 
+	// Build weight map
 	for _, aminoAcid := range codontable.AminoAcids {
 		for _, codon := range aminoAcid.Codons {
```

Comment:
"For each codon related to said amino acid get the normalized codon weight ratio and add it to the weight map."

Reply:
We're not training kids to read golang, they can see that we're ranging through amino acids and codons cause its right there, in fewer words than the comment. Will add the thing about adding to weight map.